src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dns/ClusterDnsCache.scala

```diff
@@ -0,0 +1,44 @@
+package org.broadinstitute.dsde.workbench.leonardo.dns
+
+import akka.actor.ActorSystem
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.leonardo.config.ProxyConfig
+import org.broadinstitute.dsde.workbench.leonardo.db.DbReference
+import org.broadinstitute.dsde.workbench.leonardo.model.Cluster
+import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.GoogleProject
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Created by rtitle on 8/25/17.
+  */
+object ClusterDnsCache extends LazyLogging {
```

Comment:
I might make this an Actor. Also https://broadinstitute.atlassian.net/browse/GAWB-2498 could maybe fit in here.